server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/jsieve/VacationReply.java

```diff
@@ -126,13 +128,13 @@ private Multipart generateNotificationContentFromReasonString() throws Messaging
             return multipart;
         }
 
-        private MailAddress retrieveOriginalSender() throws AddressException {
+        private MailAddress retrieveOriginalSender() {
             return Optional.ofNullable(from)
-                .map(address -> retrieveAddressFromString(address, context))
+                .map(this::retrieveAddressFromString)
```

Comment:
this function can return null, in this case this function will also return null, no?

Reply:
Yes we should do a flatMap